tests/functional/test_create.py

```diff
@@ -0,0 +1,267 @@
+
+import pytest
+import scripttest
+
+IS_WINDOWS = (
+    sys.platform.startswith("win") or
+    (sys.platform == "cli" and os.name == "nt")
+)
+IS_26 = sys.version_info[:2] == (2, 6)
+IS_PYPY = hasattr(sys, 'pypy_version_info')
+
+
+def locate_on_path(binary):
+    paths = os.environ["PATH"].split(os.path.pathsep)
+    for path in paths:
+        binpath = os.path.join(path, binary)
+        if os.path.exists(binpath):
+            return binpath
+
+
+def resolve_path(path):
+    if path is not None:
+        path = os.path.realpath(os.path.expanduser(path))
+    return path
+
+
+PYTHON_BINS = [
+    (True, "C:\\Python27\\python.exe", "lib\\site-packages"),
+    (True, "C:\\Python27-x64\\python.exe", "lib\\site-packages"),
+    (True, "C:\\Python33\\python.exe", "lib\\site-packages"),
+    (True, "C:\\Python33-x64\\python.exe", "lib\\site-packages"),
+    (True, "C:\\Python34\\python.exe", "lib\\site-packages"),
+    (True, "C:\\Python34-x64\\python.exe", "lib\\site-packages"),
+    (True, "C:\\PyPy\\pypy.exe", "site-packages"),
+    (True, "C:\\PyPy3\\pypy.exe", "site-packages"),
+    (False, None, None),
+    (True, resolve_path("~/.pyenv/shims/python"), None),
+    (True, resolve_path("~/.pyenv/shims/python2.6"), "lib/python2.6/site-packages"),
+    (True, resolve_path("~/.pyenv/shims/python2.7"), "lib/python2.7/site-packages"),
+    (True, resolve_path("~/.pyenv/shims/python3.2"), "lib/python3.2/site-packages"),
+    (True, resolve_path("~/.pyenv/shims/python3.3"), "lib/python3.3/site-packages"),
+    (True, resolve_path("~/.pyenv/shims/python3.4"), "lib/python3.4/site-packages"),
+    (True, resolve_path("~/.pyenv/shims/pypy"), "site-packages"),
+]
+if not os.environ.get("CIRCLE_BUILD_NUM"):
+    # CircleCI messed these up badly ... they use pyenv
+    PYTHON_BINS += [
+        (True, "/usr/bin/python", None),
+        (True, "/usr/bin/python2.6", "lib/python2.6/site-packages"),
+        (True, "/usr/bin/python2.7", "lib/python2.7/site-packages"),
+        (True, "/usr/bin/python3.2", "lib/python3.2/site-packages"),
+        (True, "/usr/bin/python3.3", "lib/python3.3/site-packages"),
+        (True, "/usr/bin/python3.4", "lib/python3.4/site-packages"),
+        (True, "/usr/bin/pypy", "site-packages"),
+    ]
+
+for path, sitepackages in [
+    (locate_on_path("python"), None),
+    (locate_on_path("python2.6"), None),
+    (locate_on_path("python2.7"), None),
+    (locate_on_path("python3.2"), None),
+    (locate_on_path("python3.3"), None),
+    (locate_on_path("python3.4"), None),
+    (locate_on_path("pypy"), None),
+]:
+    # I'm terrible here but I want certain checks disabled for these paths: the --system-site-packages checks, otherwise
+    # I have to reimplement bin resolving here, and it's a bad idea to duplicate logic in tests.
+    if (True, path, sitepackages) not in PYTHON_BINS:
+        if path:
+            PYTHON_BINS.append((False, path, sitepackages))
+
+PYTHON_BINS = [
+    (is_global, path, sitepackages)
```

Comment:
could we apply resolve-path inconditionaly here instead of line 41-47 ? (not sure how it behave on windows, just proposing)